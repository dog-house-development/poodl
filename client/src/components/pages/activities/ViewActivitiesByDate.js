import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import moment from 'moment';

import ActivityActions from '../../../actions/activityActions';
import ViewByDate from '../../ui/ViewByDate';

export class ViewActivitiesByDate extends Component {
    constructor(props) {
        super(props);
        this.state = {
            activityDate: moment().startOf('day')
        };
    }

    getDateRangeFilter(start) {
        const filter = {
            startDate: {
                $lt: start
                    .clone()
                    .add(1, 'days')
                    .toISOString()
            },
            endDate: {
                $gt: start.toISOString()
            }
        };

        return filter;
    }

    componentDidMount() {
        this.props.activityActions.filter(this.getDateRangeFilter(this.state.activityDate));
    }

    requestDate = date => {
        this.setState({ activityDate: date });
        this.props.activityActions.filter(this.getDateRangeFilter(date));
    };

    render() {
        return (
            <div>
                <h2>Activities</h2>
                <ViewByDate
                    requestDate={this.requestDate}
                    loading={this.props.activitiesLoading}
                    dateData={{
                        date: this.state.activityDate,
                        data: this.props.activities
                    }}
                    clickableRowRoute="activities/"
                    errors={this.props.errors}
                />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        auth: state.auth,
        activities: state.activities.all,
        activitiesLoading: state.activities.loading,
        errors: state.activities.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        activityActions: bindActionCreators(ActivityActions, dispatch)
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewActivitiesByDate);
