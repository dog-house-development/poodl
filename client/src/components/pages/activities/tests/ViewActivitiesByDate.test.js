import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';
import { Link } from 'react-router-dom';

import { ViewActivitiesByDate, mapStateToProps, mapDispatchToProps } from '../ViewActivitiesByDate';

configure({ adapter: new Adapter() });

describe('View ActivitiesByDate test', () => {
    let state, props, wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                auth: {
                    admin: {
                        _id: '123',
                        firstName: 'Moose',
                        lastName: 'Man',
                        email: 'mooseman@nowhere.com',
                        superAdmin: true
                    },
                    errors: {}
                },
                activities: {
                    loading: false,
                    all: [
                        {
                            _id: '1',
                            name: 'Fun',
                            description: 'A fun time with fun activities',
                            startDate: '1971-02-01T07:00:00.000Z',
                            endDate: '1971-02-01T07:00:00.000Z',
                            members: [
                                { firstName: 'Big', lastName: 'Tup', _id: '123' },
                                { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                            ]
                        },
                        {
                            _id: '2',
                            name: 'Stuff',
                            description: 'A fun time with fun stuff',
                            startDate: '1970-01-01T07:00:00.000Z',
                            endDate: '1970-01-01T07:00:00.000Z',
                            members: [
                                { firstName: 'Big', lastName: 'Tup', _id: '123' },
                                { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                            ]
                        }
                    ],
                    errors: {}
                }
            },
            _state
        );
        props = _.assign({}, _props);
        wrapper = shallow(
            <ViewActivitiesByDate
                {..._.assign(
                    {},
                    props,
                    mapStateToProps(state, props),
                    mapDispatchToProps(jasmine.createSpy('dispatch'))
                )}
            />
        );
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                activities: [
                    {
                        _id: '1',
                        name: 'Fun',
                        description: 'A fun time with fun activities',
                        startDate: '1971-02-01T07:00:00.000Z',
                        endDate: '1971-02-01T07:00:00.000Z',
                        members: [
                            { firstName: 'Big', lastName: 'Tup', _id: '123' },
                            { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                        ]
                    },
                    {
                        _id: '2',
                        name: 'Stuff',
                        description: 'A fun time with fun stuff',
                        startDate: '1970-01-01T07:00:00.000Z',
                        endDate: '1970-01-01T07:00:00.000Z',
                        members: [
                            { firstName: 'Big', lastName: 'Tup', _id: '123' },
                            { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                        ]
                    }
                ],
                activitiesLoading: false,
                auth: {
                    admin: {
                        _id: '123',
                        firstName: 'Moose',
                        lastName: 'Man',
                        email: 'mooseman@nowhere.com',
                        superAdmin: true
                    },
                    errors: {}
                },
                errors: {}
            });
        });
    });

    describe('mapDispatchToProps', () => {
        it('should map dispatch to props', () => {
            const dispatch = jest.fn();
            expect(JSON.stringify(mapDispatchToProps(dispatch))).toEqual(JSON.stringify({ activityActions: {} }));
        });
    });

    describe('componentDidMount', () => {
        it('should run without breaking', () => {
            spyOn(instance, 'componentDidMount');
            instance.componentDidMount();
        });
    });

    describe('render', () => {
        xit('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
