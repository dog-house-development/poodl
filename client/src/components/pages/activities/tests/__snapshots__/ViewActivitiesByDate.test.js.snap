// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View ActivitiesByDate test render should render correctly 1`] = `
<div>
  <h2>
    Activities
  </h2>
  <withRouter(ViewByDate)
    clickableRowRoute="activities/"
    dateData={
      Object {
        "data": Array [
          Object {
            "_id": "1",
            "description": "A fun time with fun activities",
            "endDate": "1971-02-01T07:00:00.000Z",
            "members": Array [
              Object {
                "_id": "123",
                "firstName": "Big",
                "lastName": "Tup",
              },
              Object {
                "_id": "321",
                "firstName": "Lil",
                "lastName": "Tup",
              },
            ],
            "name": "Fun",
            "startDate": "1971-02-01T07:00:00.000Z",
          },
          Object {
            "_id": "2",
            "description": "A fun time with fun stuff",
            "endDate": "1970-01-01T07:00:00.000Z",
            "members": Array [
              Object {
                "_id": "123",
                "firstName": "Big",
                "lastName": "Tup",
              },
              Object {
                "_id": "321",
                "firstName": "Lil",
                "lastName": "Tup",
              },
            ],
            "name": "Stuff",
            "startDate": "1970-01-01T07:00:00.000Z",
          },
        ],
        "date": "2019-04-20T06:00:00.000Z",
      }
    }
    errors={Object {}}
    loading={false}
    noDataMessage="No activities scheduled for this day"
    requestDate={[Function]}
  />
</div>
`;
